assets/javascripts/modules/autoComplete.js

```diff
@@ -314,28 +323,99 @@ var suggestionsEvent = function () {
   });
 };
 
+
+/**
+  * Generate random variable name to ensure no conflicts
+*/
+var randomVariableName = function () {
+  var text = "suggestions";
+  var possible = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+  for( var i=0; i < 5; i++ ) {
+      text += possible.charAt(Math.floor(Math.random() * possible.length));
+  }
+  return text;
+};
+
+/**
+  * Either grab a given global variable or generate one from a given select input
+*/
+var getSuggestions = function (el) {
+  var suggestions;
+  var dataLinked = $('select[id="' + el.attr("data-linked") + '"]');
+  if(dataLinked.length > 0){
+    // using a select input in-page as the source for the suggestions
+    // this is useful when using a select as a fallback for the autocomplete
+    suggestions = [];
+    dataLinked.find('option').each(function(){
+      if($(this).text() > ""){
+        suggestions.push({"title":$(this).text(), "value":$(this).attr("value")});
+      }
+    });
+    // need to populate the data-suggestions attribute from this for validation
+    // generate a global variable to hold the data and insert into data-suggestions attribute
+    var r = randomVariableName();
+    window[r] = suggestions;
+    el.attr("data-suggestions", r);
+  } else {
+    // allow custom variable
+    // this works in the same way as the legacy version by passing in an existing global object name
+    suggestions = window[el.attr("data-suggestions")];
+  }
+  return suggestions;
+};
+
+/**
+  * Find input to post value back to
+*/
+var getTargetInput = function (el) {
+  var target;
+  // input to post back a selection from the autocomplete
+  var dataPostTo = $('[id="' + el.attr("data-post-to") + '"]');
```

Comment:
`data-target-elem` feels a little bit more descriptive when looking at it in the html. `data-post-to` feels a bit `POST` form'y ... thoughts?